types/bluebird/index.d.ts

```diff
@@ -58,6 +58,12 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
    * The new promise will be rejected or resolved depending on the passed `fulfilledHandler`, `rejectedHandler` and the state of this promise.
    */
   // Based on PromiseLike.then, but returns a Bluebird instance.
+  // to support static .all
+  then<T1, T2>(onFulfill?: (value: R) => Resolvable<[T1, T2]>, onReject?: (error: any) => Resolvable<[T1, T2]>): Bluebird<[T1, T2]>;
+  then<T1, T2, T3>(onFulfill?: (value: R) => Resolvable<[T1, T2, T3]>, onReject?: (error: any) => Resolvable<[T1, T2, T3]>): Bluebird<[T1, T2, T3]>;
+  then<T1, T2, T3, T4>(onFulfill?: (value: R) => Resolvable<[T1, T2, T3, T4]>, onReject?: (error: any) => Resolvable<[T1, T2, T3, T4]>): Bluebird<[T1, T2, T3, T4]>;
+  then<T1, T2, T3, T4, T5>(onFulfill?: (value: R) => Resolvable<[T1, T2, T3, T4, T5]>, onReject?: (error: any) => Resolvable<[T1, T2, T3, T4, T5]>): Bluebird<[T1, T2, T3, T4, T5]>;
```

Comment:
* Could you please order these overloads the same way as [`.all`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/795ce172038dbafcb9cba030d637d733a7eea19c/types/bluebird/index.d.ts#L569-L574) is overloaded? That is: Descending from the T1-T5 variant down to the T1-T2 one.
* Additionally, to make it consistent, I'd like to ask you to please rename `<U>` to `<T1>` below.
* The empty newline L66 can also be removed.